### DISCUSSION TITLE
Wayland support?

### DISCUSSION THREAD

**Discussion by nathan-a-macleod** [BuddiesOfBudgie/budgie-desktop#324] [Q&A] [answered]
I've just tried out budgie and it looks great. I'm wondering if it currently supports wayland or if wayland support is planned?

**Comment by EbonJaeger** [comment, marked as answer]
Wayland support is planned for the rewrite in Budgie 11.

**Comment by fossfreedom** [comment]
Wayland support could in theory be added in the 10.x series if someone had the energy to-do.  Talking to the GNOME devs at a conference in Prague they thought the budgie ability to have panels on each edge of the screen is now possible with mutter.  Certainly gnome-shell extensions can achieve that.

The budgie panel would though need to have two ways of working - one working with xorg and its "struts" method - and another how-ever mutter+gnome-shell extensions deal with it.

Saying that - EbonJaeger is quite right to say this is a budgie 11 plan so any effort here in 10.x is kind of "throw away" in some sense - in another sense it could inform the sort of things a budgie panel needs to worry about to be fully wayland compatible.

**Comment by nathan-a-macleod** [comment]
And is the plan to use mutter as the basic wayland compositor? What about other ones such as weston?

**Reply by JoshStrobl** [reply]
No, GNOME technology will not be used for Budgie 11. Budgie 11 will use either wlroots or a wlroots inspired compositor as its base, such as smithay.

**Reply by VarLad** [reply]
@JoshStrobl There's also https://github.com/mahkoh/jay
